### PR TITLE
LSIF: Remove tables from schema

### LINKED ISSUE #45083: RFC 740: Remove LSIF tables
Blocked by https://github.com/sourcegraph/sourcegraph/issues/45081 and https://github.com/sourcegraph/sourcegraph/issues/45082.

Remove the LSIF tables from the database.

### PR DESCRIPTION
**Pull request by efritz**
Fixes #45083. This removes all LSIF-specific data from codeintel-db.

## Test plan

Main-dry-run.

### CODE REVIEW

**Review by Numbers88s — APPROVED** [review]
:letsgoooo:

**Review by Strum355 — APPROVED** [review]
No cascade constraints thatll accidentally delete stuff we want?

**Comment by efritz** [comment]
> No cascade constraints thatll accidentally delete stuff we want?

BETTER NOT BE from lsif_data* tables.

**Comment by efritz** [comment]
Fixing oobmigration errors in #47964.

File: migrations/codeintel/1676423214_remove_lsif_data/down.sql [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+CREATE INDEX IF NOT EXISTS lsif_data_documents_schema_versions_dump_id_schema_version_boun ON lsif_data_documents_schema_versions USING btree (dump_id, min_schema_version, max_schema_version);
+CREATE INDEX IF NOT EXISTS lsif_data_implementations_dump_id_schema_version ON lsif_data_implementations USING btree (dump_id, schema_version);
+CREATE INDEX IF NOT EXISTS lsif_data_implementations_schema_versions_dump_id_schema_versio ON lsif_data_implementations_schema_versions USING btree (dump_id, min_schema_version, max_schema_version);
+CREATE INDEX IF NOT EXISTS lsif_data_references_dump_id_schema_version ON lsif_data_references USING btree (dump_id, schema_version);
+CREATE INDEX IF NOT EXISTS lsif_data_references_schema_versions_dump_id_schema_version_bou ON lsif_data_references_schema_versions USING btree (dump_id, min_schema_version, max_schema_version);
+
+CREATE OR REPLACE TRIGGER lsif_data_definitions_schema_versions_insert AFTER INSERT ON lsif_data_definitions REFERENCING NEW TABLE AS newtab FOR EACH STATEMENT EXECUTE FUNCTION update_lsif_data_definitions_schema_versions_insert();
```
Comment (eseliger) [inline migrations/codeintel/1676423214_remove_lsif_data/down.sql:160]: databases cap says no 
```suggestion
DROP TRIGGER lsif_data_definitions_schema_versions_insert ON lsif_data_definitions;
CREATE TRIGGER lsif_data_definitions_schema_versions_insert AFTER INSERT ON lsif_data_definitions REFERENCING NEW TABLE AS newtab FOR EACH STATEMENT EXECUTE FUNCTION update_lsif_data_definitions_schema_versions_insert();
```

Comment (efritz) [reply migrations/codeintel/1676423214_remove_lsif_data/down.sql]: [bcd7434](https://github.com/sourcegraph/sourcegraph/pull/46127/commits/bcd743440227e335cb08c22c7872d0d36523cc15)

File: internal/database/migration/shared/data/frozen/3.34.2.json [hunk 2]
```diff
@@ -1,2089 +0,0 @@
-{
```
Comment (eseliger) [inline internal/database/migration/shared/data/frozen/3.34.2.json:1]: was this intentionally deleted?

**Review by eseliger — APPROVED** [review]

File: internal/database/migration/shared/data/frozen/3.34.2.json (hunk 2 above)
Comment (efritz) [reply internal/database/migration/shared/data/frozen/3.34.2.json:1]: Yes, I've moved to to a more recent version so that it's testing the EoL schema.

**Pull request merged**